src/codegen/yul/builtin.rs

```diff
@@ -0,0 +1,513 @@
+
+            let main_expr = if matches!(builtin_ty, YulBuiltInFunction::AddMod) {
+                Expression::Add(*loc, left.ty(), false, Box::new(left), Box::new(right))
+            } else {
+                Expression::Multiply(*loc, left.ty(), false, Box::new(left), Box::new(right))
+            };
+
+            let mod_arg = expression(&args[2], contract_no, ns, vartab, cfg, opt);
+            let (mod_left, mod_right) = equalize_types(main_expr, mod_arg, ns);
+            let codegen_expr = Expression::UnsignedModulo(*loc, mod_left.ty(), Box::new(mod_left), Box::new(mod_right.clone()));
+            branch_if_zero(mod_right, codegen_expr, ns, cfg, vartab)
+        }
+
+        YulBuiltInFunction::SignExtend
+        | YulBuiltInFunction::Keccak256
+        | YulBuiltInFunction::Pop
+        | YulBuiltInFunction::Pc
+        | YulBuiltInFunction::ChainId
+        | YulBuiltInFunction::BaseFee
+        // Memory functions: need to convert between number to pointer type
+        | YulBuiltInFunction::MLoad
+        | YulBuiltInFunction::MStore
+        | YulBuiltInFunction::MStore8
+        | YulBuiltInFunction::MSize
+        // Storage function: need to think about how to deal with pointer size and the size of chunk to load
+        | YulBuiltInFunction::SStore
+        | YulBuiltInFunction::SLoad
+        // Calldata functions: the same problems with other memory functions
+        | YulBuiltInFunction::CallDataLoad
+        | YulBuiltInFunction::CallDataSize
+        | YulBuiltInFunction::CallDataCopy
+        // Functions that manage code memory
+        | YulBuiltInFunction::CodeSize
+        | YulBuiltInFunction::CodeCopy
+        | YulBuiltInFunction::ExtCodeSize
+        | YulBuiltInFunction::ExtCodeCopy
+        | YulBuiltInFunction::ExtCodeHash
+        // Functions that manage return data
+        | YulBuiltInFunction::ReturnDataSize
+        | YulBuiltInFunction::ReturnDataCopy
+        // Functions that manage contracts
+        | YulBuiltInFunction::Create
+        | YulBuiltInFunction::Create2
+        | YulBuiltInFunction::Call
+        | YulBuiltInFunction::CallCode
+        | YulBuiltInFunction::DelegateCall
+        | YulBuiltInFunction::StaticCall
+        // Return and revert also load from memory, so we first need to solve mload and mstore builtins
+        | YulBuiltInFunction::Return
+        | YulBuiltInFunction::Stop // Stop is the same as return(0, 0)
+        | YulBuiltInFunction::Revert
+        // Log functions
+        | YulBuiltInFunction::Log0
+        | YulBuiltInFunction::Log1
+        | YulBuiltInFunction::Log2
+        | YulBuiltInFunction::Log3
+        | YulBuiltInFunction::Log4
+        // origin is the same as tx.origin and is not implemented
+        | YulBuiltInFunction::Origin
+        => {
+            let function_ty = builtin_ty.get_prototype_info();
+            unimplemented!("{} yul builtin not implemented", function_ty.name);
```

Comment:
please don't make solang crash

Reply:
Solang will never crash here, because there is an error in sema.

Reply:
should it be `unreachable!(..)` in that case?

I read `unimplemented!()` as "some code should be inserted here, developer has not finished their job"